aws/data_source_aws_codestarconnections_connection_test.go

```diff
@@ -0,0 +1,84 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/codestarconnections"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccDataSourceAwsCodeStarConnectionsConnection_basic(t *testing.T) {
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "data.aws_codestarconnections_connection.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck(codestarconnections.EndpointsID, t) },
+		Providers: testAccProviders,
```

Comment:
We are currently in the process of adding `ErrorCheck` to each of the acceptance tests. Reference: https://github.com/hashicorp/terraform-provider-aws/issues/18175

Can you please add this here? Thank you!

```suggestion
		PreCheck:   func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck(codestarconnections.EndpointsID, t) },
		ErrorCheck: testAccErrorCheck(t, codestarconnections.EndpointsID),
		Providers:  testAccProviders,
```